core/templates/components/state-editor/state-hints-editor/state-hints-editor.component.ts

```diff
@@ -13,231 +13,217 @@
+import { StateNextContentIdIndexService } from 'components/state-editor/state-editor-properties-services/state-next-content-id-index.service';
+import { StateSolutionService } from 'components/state-editor/state-editor-properties-services/state-solution.service';
+import { FormatRtePreviewPipe } from 'filters/format-rte-preview.pipe';
+import { AlertsService } from 'services/alerts.service';
+import { EditabilityService } from 'services/editability.service';
+import { WindowDimensionsService } from 'services/contextual/window-dimensions.service';
+import { ExternalSaveService } from 'services/external-save.service';
 import { AddHintModalComponent } from 'pages/exploration-editor-page/editor-tab/templates/modal-templates/add-hint-modal.component';
+import { DeleteHintModalComponent } from 'pages/exploration-editor-page/editor-tab/templates/modal-templates/delete-hint-modal.component';
+import { DeleteLastHintModalComponent } from 'pages/exploration-editor-page/editor-tab/templates/modal-templates/delete-last-hint-modal.component';
+import { NgbModal } from '@ng-bootstrap/ng-bootstrap';
+import { CdkDragSortEvent, moveItemInArray} from '@angular/cdk/drag-drop';
+
+@Component({
+  selector: 'oppia-state-hints-editor',
+  templateUrl: './state-hints-editor.component.html'
+})
+export class StateHintsEditorComponent implements OnInit {
+  @Input() onSaveHints;
+  @Input() onSaveNextContentIdIndex;
+  @Input() onSaveSolution;
+  @Input() showMarkAllAudioAsNeedingUpdateModalIfRequired;
+  hintCardIsShown: boolean;
+  canEdit: boolean;
+
+  constructor(
+    private stateHintsService: StateHintsService,
+    private urlInterpolationService: UrlInterpolationService,
+    private stateEditorService: StateEditorService,
+    private stateInteractionIdService: StateInteractionIdService,
+    private stateNextContentIdIndexService: StateNextContentIdIndexService,
+    private stateSolutionService: StateSolutionService,
+    private alertsService: AlertsService,
+    private editabilityService: EditabilityService,
+    private windowDimensionsService: WindowDimensionsService,
+    private externalSaveService: ExternalSaveService,
+    private formatRtePreviewPipe: FormatRtePreviewPipe,
+    private ngbModal: NgbModal,
+  ) {}
+
+  drop(event: CdkDragSortEvent<Hint[]>): void {
+    moveItemInArray(
+      this.stateHintsService.displayed, event.previousIndex,
+      event.currentIndex);
+    this.stateHintsService.saveDisplayedValue();
+    this.onSaveHints(this.stateHintsService.displayed);
+  }
+
+  getHintButtonText(): string {
+    var hintButtonText = '+ ADD HINT';
+    if (this.stateHintsService.displayed) {
+      if (this.stateHintsService.displayed.length >= 5) {
+        hintButtonText = 'Limit Reached';
+      }
+    }
+    return hintButtonText;
+  }
+
+  getHintSummary(hint: Hint): string {
+    const hintAsPlainText =
+      this.formatRtePreviewPipe.transform(hint.hintContent.html);
```

Comment:
Please wrap in parenthesis when breaking a statement into multiple lines.

Reply:
Done